lib/app/services/api_network.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/widgets.dart';
+import 'package:get_it/get_it.dart';
+import 'package:meta/meta.dart';
+import 'package:http/http.dart' as http;
+import 'package:schulcloud/app/app.dart';
+
+import 'storage.dart';
+
+/// A service that offers networking POST and GET requests to the backend
+/// servers. If the user's token is stored in the authentication storage, the
+/// the requests' headers are automatically enriched with the access token.
+@immutable
+class ApiNetworkService {
+  const ApiNetworkService({@required this.apiUrl}) : assert(apiUrl != null);
```

Comment:
Why use a constructor parameter for the URL? You can retrieve it as `services.get<AppConfig>.baseApiUrl`